### PR TITLE
Replace SchemaCleaner with ConnectionHelper

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/issues/16021

This disables constraints before dropping and truncating tables.

### CODE REVIEW

File: src/TestSuite/Fixture/SchemaLoader.php [hunk 1]
```diff
@@ -61,37 +55,33 @@ class SchemaLoader
     public function __construct(array $config = [])
```
Comment (othercorey) [inline src/TestSuite/Fixture/SchemaLoader.php]: Think we could drop the constructor config. Not sure if users would need global options here.

File: src/TestSuite/ConnectionHelper.php [hunk 2]
```diff
@@ -70,4 +72,80 @@ public function enableQueryLogging(array $connections = []): void
...
+     * @param array|null $tables List of tables names or null for all.
+     * @return void
+     */
+    public function dropTables(string $connection, ?array $tables = null): void
+    {
+        /** @var \Cake\Database\Connection $connection */
+        $connection = ConnectionManager::get($connection);
+        $collection = $connection->getSchemaCollection();
+
+        $allTables = $connection->getSchemaCollection()->listTables();
```
Comment (markstory) [inline src/TestSuite/ConnectionHelper.php]:
```suggestion
        $allTables = $collection->listTables();
```

File: src/TestSuite/ConnectionHelper.php [hunk 3]
```diff
@@ -70,4 +72,80 @@ public function enableQueryLogging(array $connections = []): void
...
+     * @param string $connection Connection name
+     * @return void
+     */
+    public function truncateTables(string $connection): void
+    {
+        /** @var \Cake\Database\Connection $connection */
+        $connection = ConnectionManager::get($connection);
+        $collection = $connection->getSchemaCollection();
+
+        $tables = $connection->getSchemaCollection()->listTables();
```
Comment (markstory) [inline src/TestSuite/ConnectionHelper.php]:
```suggestion
        $tables = $collection->listTables();
```

File: src/TestSuite/Fixture/SchemaCleaner.php [hunk 4]
```diff
@@ -1,171 +0,0 @@
...
-use Cake\Database\Schema\SqlGeneratorInterface;
-use Cake\Datasource\ConnectionInterface;
-use Cake\Datasource\ConnectionManager;
-
-/**
- * This class will help dropping and truncating all tables of a given connection
- *
- * @internal
- */
-class SchemaCleaner
```
Comment (markstory) [inline src/TestSuite/Fixture/SchemaCleaner.php:30]: We'll need to update migrations for this as well.

File: src/TestSuite/ConnectionHelper.php (hunk 2 above)
Comment (othercorey) [reply src/TestSuite/ConnectionHelper.php]: The dangers of moving code around.

File: src/TestSuite/Fixture/SchemaCleaner.php (hunk 4 above)
Comment (othercorey) [reply src/TestSuite/Fixture/SchemaCleaner.php:30]: Will do.

**Comment by othercorey** [comment]
This is ready now. We drop constraints on all tables before dropping the tables.

I tried to describe tables the minimum amount as I hope most users aren't caching their test connection.

**Pull request merged**